Source/Core/DolphinQt/Debugger/CodeDiffDialog.cpp

```diff
@@ -0,0 +1,406 @@
+    return;
+  }
+
+  std::vector<Diff> current_diff;
+  if (m_exclude.empty())
+  {
+    current_diff = recorded_symbols;
+  }
+  else
+  {
+    for (auto& iter : recorded_symbols)
+    {
+      auto pos = lower_bound(m_exclude.begin(), m_exclude.end(), iter.symbol, AddrOP);
+
+      if (pos->symbol != iter.symbol)
+      {
+        current_diff.push_back(iter);
+      }
+    }
+  }
+
+  if (!m_include.empty())
+  {
+    m_include.erase(std::remove_if(m_include.begin(), m_include.end(),
+                                   [&](Diff& v) {
+                                     // Remove values if it can't find matching symbols
+                                     return std::none_of(
+                                         current_diff.begin(), current_diff.end(),
+                                         [&](Diff& p) { return p.symbol == v.symbol; });
+                                   }),
+                    m_include.end());
+  }
+  else
+  {
+    m_include = recorded_symbols;
+    RemoveMatchingSymbolsFromIncludes(m_exclude);
+  }
+}
+
+void CodeDiffDialog::OnExclude()
+{
+  const auto recorded_symbols = CalculateSymbolsFromProfile();
+  if (m_include.empty() && m_exclude.empty())
+  {
+    m_exclude = recorded_symbols;
+    return;
+  }
+
+  std::vector<Diff> current_diff;
+  if (m_exclude.empty())
+  {
+    m_exclude = recorded_symbols;
+  }
+  else
+  {
+    for (auto& iter : recorded_symbols)
+    {
+      auto pos = lower_bound(m_exclude.begin(), m_exclude.end(), iter.symbol, AddrOP);
+
+      if (pos->symbol != iter.symbol)
+      {
+        current_diff.push_back(iter);
+
+        m_exclude.insert(pos, iter);
+      }
+    }
+    // If there is no include list, we're done.
+    if (m_include.empty())
+      return;
+  }
+
+  if (!m_exclude.empty())
+  {
+    RemoveMatchingSymbolsFromIncludes(current_diff);
+  }
+}
+
+std::vector<Diff> CodeDiffDialog::CalculateSymbolsFromProfile()
+{
+  Profiler::ProfileStats prof_stats;
+  auto& blockstats = prof_stats.block_stats;
+  JitInterface::GetProfileResults(&prof_stats);
+  std::vector<Diff> current;
+  current.reserve(20000);
+
+  // Convert blockstats to smaller struct Diff. Exclude repeat functions via symbols.
+  for (auto& iter : blockstats)
+  {
+    Diff tmp_diff;
+    std::string symbol = g_symbolDB.GetDescription(iter.addr);
+    if (!std::any_of(current.begin(), current.end(),
+                     [&symbol](Diff& v) { return v.symbol == symbol; }))
+    {
+      tmp_diff.symbol = symbol;
+      tmp_diff.addr = iter.addr;
+      tmp_diff.hits = iter.run_count;
+      current.push_back(tmp_diff);
+    }
+  }
+
+  // Could add address based difference instead of symbols. Probably need second function.
+  // Sort for lower_bound.
+  sort(current.begin(), current.end(),
+       [](const Diff& v1, const Diff& v2) { return (v1.symbol < v2.symbol); });
+
+  return current;
+}
+
+void CodeDiffDialog::RemoveMatchingSymbolsFromIncludes(std::vector<Diff>& symbol_list)

```

Comment:
The parameter should be passed as const reference:
```c++
void CodeDiffDialog::RemoveMatchingSymbolsFromIncludes(const std::vector<Diff>& symbol_list)
```

Reply:
Violates algorithm.h, not changing

Reply:
That's odd, I'm pretty sure it shouldn't. Have you changed the lambdas' parameter to take const references as well?

Basically, you need to replace occurrences of `[&](Diff& v)` with`[&](const Diff& v)`.

Reply:
... well this is embarrassing.
I was treating the lambdas as separate things. I did not realize they needed to match. But I realize now it has to match otherwise the type is wrong when iterating.

Added const where appropriate, done.